### PR TITLE
Renames land BGC files

### PR DESCRIPTION
**Pull request by bishtgautam**
Renames land biogeochemistry files to follow a consistent naming convention.

[BFB]

### CODE REVIEW

File: components/clm/src/biogeochem/CH4Mod.F90 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module ch4Mod
+module CH4Mod
```
Comment (bpbond) [inline components/clm/src/biogeochem/CH4Mod.F90]: 👍

File: components/clm/src/biogeochem/CNEcosystemDynBetrMod.F90 [hunk 2]
```diff
@@ -328,35 +328,35 @@ subroutine CNEcosystemDynBetr(bounds,                             &
        call t_startf('CNUpdate1')
 
        if ( use_c13 ) then
-          call CIsoFlux1(num_soilc, filter_soilc, num_soilp, filter_soilp, &
+          call CarbonIsoFlux1(num_soilc, filter_soilc, num_soilp, filter_soilp, &
```
Comment (bpbond) [inline components/clm/src/biogeochem/CNEcosystemDynBetrMod.F90]: Lots clearer!

**Comment by bishtgautam** [comment]
@bpbond, I have rebased this branch off the latest master and have renamed additional files as we discussed during last week's meeting. When you get a chance, please review it again. Thanks.

**Comment by jonbob** [comment]
@bishtgautam - is this something we need for maint-1.1?

**Comment by bishtgautam** [comment]
@jonbob, This PR is not needed for maint-1.1.

**Pull request merged**